libs/base/Data/List.idr

```diff
@@ -36,3 +36,17 @@ isElem x (y :: xs) with (decEq x y)
                Elem x' (y' :: xs') -> Void
         mkNo f g Here = f Refl
         mkNo f g (There x) = g x
+
+intersectBy : (a -> a -> Bool) -> List a -> List a -> List a
```

Comment:
@david-christiansen I followed `unionBy` which has no documentation also, but I can add it to both, `[]` cases reason is same as for `unionBy`, it's more interesting for me why I should not add it.